A41-xds-rbac.md

```diff
@@ -0,0 +1,199 @@
+values received on-the-wire independent of whether they are stored in Metadata
+or in separate APIs. It is unspecified whether hop-to-hop headers are matched.
+Multi-valued metadata is represented as the concatenation of the values along
+with a `,` (comma, no added spaces) separator, as permitted by HTTP and gRPC.
+Binary headers are represented in their base64-encoded form, although we rarely
+expect binary header matchers.
+
+TODO: Clearly define differences compared to client-side. Align with client-side
+routing (maybe changing behavior) to allow reusing code, have a strong grip on
+client vs server differences, and notice 'gotcha's.  Content-Type is probably
+not a problem on server-side.
+
+As documented for `HeaderMatcher`, Envoy aliases `:authority` and `Host` in its
+header map implementation, so they should be treated equivalent for the RBAC
+matchers; there must be no behavior change depending on which of the two header
+names is used in the _RBAC policy_.
+
+TODO: We've had no reason to care about :scheme in our APIs and so is not
+present in all languages. It could be inferred from whether the connection used
+security, but that's technically different from :scheme, as if you go through a
+TLS-terminating proxy and use plaintext to the backend the :scheme is used to
+re-create the original URL which would be 'https', not 'http'. This may be a
+problem for Java and Go. We can maybe copy whatever Envoy does for HTTP/1
+traffic. You might think :scheme doesn't matter, and maybe it doesn't, but it
+might be used to detect "is the client using TLS". Although note that in that
+case such a check is trusting the client to provide the correct value.
+
+TODO: Need to audit that server implementations check that :method=POST before
+accepting the RPC. This was already a requirement but it is an appropriate time
+to double-check.
+
+TODO: Determine how to handle _requests_ (not policies) that have Host header,
+since gRPC generally doesn't observe this header yet the RBAC policy could.
+Options: deny such requests, move Host header to :authority, drop Host header.
+Host and :authority could also disagree, and that needs to be handled.
+
+TODO: Should MetadataMatcher's NullMatch match? Need to check not present vs
+present-but-null behavior.
+
+Note that `requested_server_name` can match if the matcher accepts empty string.
+TODO: does Envoy default to empty string or null?
+
+The following fields of `Principal` may not be obvious how they map to gRPC:
+
+| Principal Field  | gRPC Equivalent |
+| ---------------- | --------------- |
+| authenticated.principal_name | The first URI/DNS SAN; same as Envoy |
+| source_ip        | Peer address for this connection |
+| direct_remote_ip | Peer address for this connection |
+| remote_ip        | Peer address for this connection |
+| header           | Metadata (with caveats) |
+| url_path         | Fully-qualified RPC method name with leading slash |
```

Comment:
Is that the same as the `url_path` in the `Permission` proto? Otherwise the principal's url_path is not known, right?